src/Init/Lean/Data/Json/Basic.lean

```diff
@@ -0,0 +1,148 @@
+/-
+Copyright (c) 2019 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+
+Authors: Gabriel Ebner, Marc Huisinga
+-/
+prelude
+import Init.System.IO
+import Init.Data.String
+import Init.Data.Int
+import Init.Data.RBMap
+
+namespace Lean
+
+-- mantissa * 10^-exponent
+structure JsonNumber :=
+(mantissa : Int) (exponent : Nat)
+
+protected def JsonNumber.decEq : ∀ (a b : JsonNumber), Decidable (a = b)
+| ⟨m1, e1⟩, ⟨m2, e2⟩ =>
+  match decEq m1 m2 with
+  | isTrue hm =>
+    match decEq e1 e2 with
+    | isTrue he  => isTrue (hm ▸ he ▸ rfl)
+    | isFalse he => isFalse (fun h => JsonNumber.noConfusion h (fun hm he2 => he he2))
+  | isFalse hm => isFalse (fun h => JsonNumber.noConfusion h (fun hm2 he => hm hm2))
+
+instance JsonNumber.decidableEq : DecidableEq JsonNumber :=
+JsonNumber.decEq
+
+namespace JsonNumber
+
+protected def fromNat (n : Nat) : JsonNumber := ⟨n, 0⟩
+protected def fromInt (n : Int) : JsonNumber := ⟨n, 0⟩
+
+instance natToJsonNumber : HasCoe Nat JsonNumber := ⟨JsonNumber.fromNat⟩
```

Comment:
Let's put a `TODO` comment here saying that we need to switch to `Coe` after we move to the new frontend.